test/server/lib/builders/document-builder.spec.js

```diff
@@ -0,0 +1,211 @@
+
+const { expect } = require('chai');
+
+const {
+	DEFAULT_FORMAT,
+	FORMAT_ARTICLE_CLEAN_ELEMENTS,
+	FORMAT_ARTICLE_STRIP_ELEMENTS,
+	TEST: { FIXTURES_DIRECTORY },
+} = require('config');
+
+const enrich = require('../../../../server/lib/enrich');
+const DocumentBuilder = require('../../../../server/lib/builders/document-builder');
+
+const MODULE_ID =
+	path.relative(`${process.cwd()}/test`, module.id) ||
+	require(path.resolve('./package.json')).name;
+
+describe(MODULE_ID, function () {
+	const CONTENT_ID = '42ad255a-99f9-11e7-b83c-9588e51488a0';
+
+	let content;
+	let documentBuilder;
+
+	beforeEach(function () {
+		content = Object.assign({}, require(path.resolve(
+			`${FIXTURES_DIRECTORY}/content/${CONTENT_ID}.json`
+		))); // Clone Content
+		content.lang = 'en';
+		content.extension = DEFAULT_FORMAT;
+
+		const contract = {
+			allowed: {
+				rich_articles: false,
+			},
+		};
+
+		enrich(content, contract);
+
+		documentBuilder = new DocumentBuilder(content);
+	});
+
+	describe('removeElementsByTagName', function () {
+
+		FORMAT_ARTICLE_STRIP_ELEMENTS.forEach(tagName => {
+			it(`contentDocument should not contain any <${tagName} /> elements`, function () {
+				documentBuilder.removeElementsByTagName();
+				const document = documentBuilder.getDocument();
+
+				expect(document.getElementsByTagName(tagName)).to.have.property('length').and.to.equal(0);
+			});
+		});
+
+		it('returns DocumentBuilder instance', function () {
+			expect(documentBuilder.removeElementsByTagName().constructor.name).to.equal(
+				'DocumentBuilder'
+			);
+		});
+	});
+
+	describe('removeProprietaryElement', function () {
+
+		FORMAT_ARTICLE_CLEAN_ELEMENTS.forEach(tagName => {
+			it(`contentDocument should not contain any <${tagName} /> elements`, function () {
+				documentBuilder.removeProprietaryElement();
+				const document = documentBuilder.getDocument();
+
+				expect(document.getElementsByTagName(tagName)).to.have.property('length').and.to.equal(0);
+			});
+		});
+
+		it('returns DocumentBuilder instance', function () {
+			expect(documentBuilder.removeProprietaryElement().constructor.name).to.equal(
+				'DocumentBuilder'
+			);
+		});
+	});
+
+	describe('removeWhiteSpace', function () {
+
+		it('contentDocument should not contain whitespace', function () {
+			documentBuilder.removeWhiteSpace();
+			const document = documentBuilder.getDocument();
+			const hasWhiteSpace = Array.from(document.documentElement.childNodes).some((el) => el.nodeType !== 1);
+
+			expect(hasWhiteSpace).and.to.equal(false);
+		});
+
+		it('returns DocumentBuilder instance', function () {
+			expect(documentBuilder.removeWhiteSpace().constructor.name).to.equal(
+				'DocumentBuilder'
+			);
+		});
+	});
+
+	describe('removeNonSyndicatableImages', function () {
+
+		let embedsMap;
+
+		before(function(){
+			embedsMap = (content.embeds || []).reduce((map, embed) => {
+				const id = embed.id.split('/').pop();
+
+				map[id] = embed;
+				return map;
+			}, {});
```

Comment:
do we need code to generate the embedsMap? Coud we not have instead the preformatted data?

Reply:
that's actually a copy of this:
https://github.com/Financial-Times/next-syndication-api/pull/265/files#diff-c22f35a272a198e0464e87259ca459b7f0ef03d3bfcadae891ac57525ddceeadR88-R93

Reply:
Yes, need it to build the map from the `content.embeds` array. Didn't want to add the `embedMap` to the fixtures because the fixture currently mirrors the response from elasticsearch.